[Docs] Discover docs updates (#198423)
elastic/kibana: docs/discover/document-explorer.asciidoc

[[document-explorer]]

> == Customize the Discover view

Author: You're absolutely right, but I didn't want to risk breaking too many links for now to keep this PR going

> == Customize the Discover view

Reviewer: yes this makes sense, let's keep it for now

Fine tune your explorations by customizing *Discover* to bring out the the best view of your documents.

> Adjust what is displayed in Discover, and how each element is displayed.

Author: ```suggestion
```

[role="screenshot"]
image::images/hello-field.png[A view of the Discover app]

[float]
[[document-explorer-c]]

> === Hide or resize elements

Author: ```suggestion
=== Hide or resize areas
```

* You can hide and show the chart and the fields list using the available collapse and expand button in the corresponding area.

* Adjust the width and height of each area by dragging their border
to the size you want.
The size of each area is saved in your browser for the next time you open **Discover**.

[float]
[[document-explorer-customize]]
=== Modify the document table

Customize the appearance of the document table and its contents to your liking.

image:images/discover-customize-table.png[Options to customize the table in Discover]

[float]
[[document-explorer-columns]]
==== Reorder and resize the columns

> * To move a single column, open the column's contextual options, and select *Move left* or *Move right* in the available options.

Author: I think I just inverted 2 screenshots which makes the ordering weirder. I'll rather fix this

+
Column widths are stored with a saved search.  When you visualize saved searches on dashboards, the saved search appears the same as in **Discover**.

[float]
[[document-explorer-density]]
==== Customize the table density

You can adjust the density of the table from the **Display options** located in the table toolbar. This can be particularly useful when scrolling through many results.

[float]
[[document-explorer-row-height]]
==== Adjust the row height

To set the row height to one or more lines, or automatically
adjust the height to fit the contents, open the **Display options** in the table toolbar, and adjust it as you need.

You can define different settings for the header row and body rows.

[float]
[[document-explorer-sample-size]]
==== Limit the sample size

> When the number of results returned by your search query (displayed at the top of the **Documents** or **Results** tab) is greater than the value of <<advanced-options#discover-sample-size,`discover:sampleSize`>>, the number of results displayed in the table is limited to the configured value by default. You can adjust the initial sample size for searches to any number between 10 and `discover:sampleSize` from the **Display options** located in the table toolbar.

Author: ```suggestion
When the number of results returned by your search query (displayed at the top of the **Documents** or **Results** tab) is greater than the value of <<kibana-discover-settings,`discover:sampleSize`>>, the number of results displayed in the table is limited to the configured value by default. You can adjust the initial sample size for searches to any number between 10 and `discover:sampleSize` from the **Display options** located in the table toolbar.
```

+
Column widths are stored with a saved search.  When you visualize saved searches on dashboards, the saved search appears the same as in **Discover**.

[float]
[[document-explorer-density]]
==== Customize the table density

> You can adjust the density of the table from the **Display options** located at the top of the table. This can be particularly useful for reading and scanning through the table.

Author: Nice yeah. There are so many places with options that I wanted to be explicit about the positioning of the options. I need to wrap my head around those toolbars/application bars/candy bars

[float]
[[document-explorer-row-height]]
==== Adjust the row height

To set the row height to one or more lines, or automatically
adjust the height to fit the contents, open the **Display options** at the top of the table, and adjust it as you need.

You can define different settings for the header and the regular rows of the table.

[float]
[[document-explorer-sample-size]]
==== Limit the sample size

> When the number of results or documents corresponding to your data view or search query (displayed at the top of the **Documents** or **Results** tab) is greater than 500, the number of results displayed in the table is limited to 500 by default. You can adjust the number of entries initially loaded to any number between 10 and 500 from the **Display options** located at the top of the table.

Author: Nice thanks for the clarification!

On the last page of the table, a message indicates that you've reached the end of the loaded search results. From that message, you can choose to load more results to continue exploring the entries. 

> image:images/discover-downsampling.png[Downsampling in Discover]

Author: Sounds good, I'll rename it
